zomes/hc_zome_trust_atom/src/trust_graph.rs

```diff
@@ -0,0 +1,45 @@
+use hdk::prelude::*;
+
+use crate::trust_atom::*;
+use crate::trust_atom::LinkDirection;
+use crate::agents::*;
+use crate::agents::*AgentRegistry;
+
+pub fn create_trust_graph_plus_rollup(register: impl Fn(EntryHash) -> AgentRegistry, filter: Option<LinkTag>) -> ExternResult<Vec<TrustAtom>> {
+    let base_address: EntryHash = agent_info()?.agent_latest_pubkey.into();
+    let mut trust_graph: Vec<TrustAtom> = create_trust_graph_mine(filter.clone())?;
+    let mut rollup: Vec<TrustAtom> = create_rollup_atoms(register, filter)?;
+    trust_graph.append(&mut rollup);
+    Ok(trust_graph) 
+}
+
+pub fn create_trust_graph_mine(filter: Option<LinkTag>) -> ExternResult<Vec<TrustAtom>> {
+    let base_address: EntryHash = agent_info()?.agent_latest_pubkey.into();
+    let links: Vec<Link> = get_links(base_address.clone(), filter.clone())?;
+    let trust_atoms: Vec<TrustAtom> = convert_links_to_trust_atoms(links, &LinkDirection::Forward, &base_address)?;
+    Ok(trust_atoms)
+}
+
+fn create_rollup_atoms(registered_agents: EntryHash, filter: Option<LinkTag>) -> ExternResult<Vec<TrustAtom>> { 
```

Comment:
if multiple links to same target, same content, different values, take only the latest one.